### ISSUE TITLE
[operator-assignment] new option for async functions

### ISSUE THREAD

**Issue by Andarist** [eslint/eslint#10583]
**What rule do you want to change?**

https://eslint.org/docs/rules/operator-assignment

**Does this change cause the rule to produce more or fewer warnings?**

more

**How will the change be implemented? (New option, new default behavior, etc.)?**

new option + new default behaviour

**Please provide some example code that this change will affect:**

```js
let count = 0

async function asyncInc() {
	count += await getNumber()
}
```

**What does the rule currently do for this code?**
**What will the rule do after it's changed?**

 Currently "operator-assignment" rule has all or nothing behaviour. I'd like to make it configurable - I find this rule too restrictive for sync code (so I dont want to use it) but I would like to forbid those kind of assignments for async functions (ideally only for those that mutate non-local variables). Thus rule with the mentioned new option could become part of `eslint:recommended`.

 **Why?**

Because `count += await getNumber()` is actually equivalent of `count = count + await getNumber()`, so it's easy to update stale value.

https://www.youtube.com/watch?v=bfxglBVSNDI

**Comment by Andarist** [comment]
Kinda a duplicate of https://github.com/eslint/eslint/issues/10405 (just found it).
